### PR TITLE
Make default security model none on MacOS

### PR DESCRIPTION
**Pull request by ssbarnea**
Related: https://github.com/containers/podman/discussions/16102

#### Does this PR introduce a user-facing change?

```release-note
Changes default security mode for podman machine to none on MacOS
in order to avoid having broken symlinks.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude @n1hility @ashley-cui PTAL

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -306,6 +307,10 @@ func (v *MachineVM) Init(opts machine.InitOptions) (bool, error) {
 		target := source
 		readonly := false
 		securityModel := "mapped-xattr"
+		if runtime.GOOS == "darwin" {
+			// see https://github.com/containers/podman/discussions/16102
+			securityModel = "none"
```
Comment (n1hility) [inline pkg/machine/qemu/machine.go:312]: IMO this should be consistent between mac and linux, so i would remove the runtime check here.

Comment (n1hility) [inline pkg/machine/qemu/machine.go:312]: IMO this should be consistent between mac and linux, so i would remove the runtime check here.

**Review by n1hility — CHANGES_REQUESTED** [review]

**Comment by n1hility** [comment]
Aside from the note on the runtime OS check, I agree with the default change. I think it better fits the workflow for container access patterns. I would check with @baude as well though.

**Comment by rhatdan** [comment]
Should this be defined in containers.conf?

File: docs/source/markdown/podman-machine-init.1.md [hunk 2]
```diff
@@ -97,8 +97,12 @@ options are:
 
 The 9p security model [determines] https://wiki.qemu.org/Documentation/9psetup#Starting_the_Guest_directly
 if and how the 9p filesystem translates some filesystem operations before
-actual storage on the host. The
-default value of *mapped-xattr* specifies that 9p store symlinks and some file
+actual storage on the host.
+
+In order to allow symlinks to work, on MacOS the default security model is
+ *none*, while on the other platforms being *mapped-xattr*.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-machine-init.1.md:103]:
```suggestion
 *none*, while on other platforms the default is *mapped-xattr*.
```

File: pkg/machine/qemu/machine.go (hunk 1 above)
Comment (n1hility) [reply pkg/machine/qemu/machine.go:312]: Was there a discussion on this point that I missed? Or was it unintentionally closed?

**Comment by rhatdan** [comment]
Please rebase to fix the CI System

File: pkg/machine/qemu/machine.go (hunk 1 above)
Comment (rhatdan) [reply pkg/machine/qemu/machine.go:312]: I agree and it should be configured in containers.conf.

**Pull request closed without merging**